controller/api/destination/federated_service_watcher.go

```diff
@@ -365,13 +365,17 @@ func (fs *federatedService) remoteDiscoverySubscribe(
 		subscriber.endStream,
 		fs.log,
 	)
+	if err != nil {
+		fs.log.Errorf("Failed to create endpoint translatorfor  remote disocvery service %q in cluster %s: %s", id.service.Name, id.cluster, err)
```

Comment:
```suggestion
		fs.log.Errorf("Failed to create endpoint translator for remote discovery service %q in cluster %s: %s", id.service.Name, id.cluster, err)
```

Reply:
Also, if `err != nil`, translator will be nil, which will cause a panic when calling Start() below, so I think we need to return at this point.